subsys/bluetooth/host/scan.c

```diff
@@ -639,22 +653,44 @@ void bt_hci_le_per_adv_report(struct net_buf *buf)
 	info.addr = &per_adv_sync->addr;
 	info.sid = per_adv_sync->sid;
 
-	if (evt->data_status == BT_HCI_LE_ADV_EVT_TYPE_DATA_STATUS_PARTIAL) {
-		/* Handling of incomplete reports is currently not
-		 * handled in the host. The remaining advertising
-		 * reports may therefore contain partial data.
-		 */
-		BT_WARN("Incomplete per adv report");
-	}
-
-	SYS_SLIST_FOR_EACH_CONTAINER(&pa_sync_cbs, listener, node) {
-		if (listener->recv) {
-			net_buf_simple_save(&buf->b, &state);
-
-			buf->len = evt->length;
-			listener->recv(per_adv_sync, &info, &buf->b);
-
-			net_buf_simple_restore(&buf->b, &state);
+	if (!per_adv_sync->report_truncated) {
+#if CONFIG_BT_PER_ADV_SYNC_BUF_SIZE > 0
+		if (evt->data_status == BT_HCI_LE_ADV_EVT_TYPE_DATA_STATUS_COMPLETE &&
+		    per_adv_sync->reassembly.len == 0) {
+			/* We have not received any partial data before.
+			 * This buffer can be forwarded without an extra copy.
+			 */
+			bt_hci_le_per_adv_report_recv(per_adv_sync, &buf->b, &info);
+			net_buf_simple_reset(&per_adv_sync->reassembly);
```

Comment:
Is it necessary to reset the buffer in this case?

Reply:
No, I will remove it